staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -82,26 +82,60 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, c client) error {
+	var wg sync.WaitGroup
+	errCh := make(chan error, len(c.Endpoints()))
+	retryCount := 3
+
+	retryFunc := func(ctx context.Context, c client, endpoint string, wg *sync.WaitGroup, errCh chan<- error) {
+		defer wg.Done()
+
+		for i := 0; i < retryCount; i++ {
+			err := f.clientSupportsRequestWatchProgress(ctx, c, endpoint)
+			if err == nil {
+				return
+			} else if i == retryCount-1 {
+				// Throw last error out
+				errCh <- err
+			}
+		}
+	}
+
+	for _, ep := range c.Endpoints() {
+		wg.Add(1)
+		go retryFunc(ctx, c, ep, &wg, errCh)
+	}
+	// Wait for all goroutines to finish
+	go func() {
+		wg.Wait()
+		close(errCh)
+	}()
+	// Throw first error out
+	for err := range errCh {
```

Comment:
I don't think that waiting for this should be blocking - we should return from this function and the check for support should be asynchronous.